iFixit/react-commerce
# Add featured products section

closes #1409 

First of all, apologies for the big PR! 😬 To anyone who'll review this I'll owe you a pizza!
The bulk of the changes comes from ramifications of refactoring common concepts used by the product previews cards (e.g. money, price tiers, etc).

---

❗❗❗After this PR is merged Strapi should be redeployed in production ❗❗❗

cc @sterlinghirsh @danielbeardsley @masonmcelvain 

---

## Changelog

- Added a featured products section to the page template
- Polished `ProductGrid` and `ProductGridItem` which will be shared pretty much everywhere there's a grid of products (e.g. product lists, featured products section, etc). This refactor will happen in a follow up PR, for now the new `ProductGrid` and `ProductGridItem` are being used only in the featured products section
- Adding more models to centralize the logic of common entities (e.g. `Money`)

## QA

1. Open [store home page preview](https://react-commerce-git-add-featured-products-section-ifixit.vercel.app/store)
2. Verify that the featured products section is implemented according to [mockup](https://www.figma.com/file/jRG4EyQXRDMGfyLqut08KP/iFixit---Working-files?node-id=10434%3A211428&t=1Q1yOcSyhFvpnYZq-4)

    ![Screenshot 2023-03-02 at 16 39 57](https://user-images.githubusercontent.com/4640135/222476044-3c03f5c4-6d0b-44ce-88a7-0e20903c241c.png)

3. Verify that the section is responsive

> 👁️ We're just using dummy data and it can be edited from Strapi
